pkg/cloudprovider/types.go

```diff
@@ -27,7 +27,7 @@ import (
 // CloudProvider holds contains the methods necessary in a cloud provider
 type CloudProvider interface {
 	// Create a set of nodes for each of the given constraints.
-	Create(context.Context, *v1alpha3.Provisioner, []*Packing) ([]*PackedNode, error)
+	Create(context.Context, *v1alpha3.Provisioner, *Packing) (*v1.Node, error)
```

Comment:
I'm concerned this limits the cloud providers' ability to do any optimization. We're kind of brute-forcing our way out of it at the controller level.  At the expense of making the Create interface more complicated, would it be worth making it asynchronous with a richer response than just a v1.Node. The controller could track the actual provisioning status and drive the packings down until the desired state is reached.

Reply:
One optimization that can probably happen pretty easily in the cloud provider currently is on the usual scale up case where Create will be launching a few or a bunch of the largest nodes possible. We can probably get significantly better scaling speeds for this case if we do one request for multiple instances. We should even be able to maintain a diverse capacity request since the pod density should be the same in this situation.

Reply:
I don't want us to limit that capability as we find more cases or more capabilities are introduced to improve the cloud provider.

Reply:
I completely agree with you here and played with the idea of using a callback model.

`Create(ctx, provisioner, packing, func(node v1.Node) error {})`

In the callback, we'd execute all of the bind logic.

Can we punt on this until we're ready to make more than one request at a time? I can also play with it here.

Reply:
Implemented this.

Reply:
ha love it!